docs/DeveloperGuide.adoc

```diff
@@ -837,33 +902,369 @@ Use case ends.
+3.  User enters the tag they want to remove
+4.  ContactSails removes the specific tag from all contacts who have this tag assigned to them
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no tags
++
+[none]
+** 2a1. ContactSails shows an error message
++
+Use case ends.
+
+[none]
+* 3a. User enters invalid tag
++
+[none]
+** 3a1. ContactSails shows an error message
 +
 Use case resumes at step 2.
 
-_{More to be added}_
+[discrete]
+=== Use case: Export contacts
+
+*MSS*
+
+1.  User selects contacts to export from the current list of contacts
+2.  ContactSails stores the selection and processes the contacts to export
+3.  User requests to create a file containing the selected contacts
+4.  ContactSails creates the file and prompts for save location of file
+5.  User specifies location of file to be saved
+6.  ContactSails saves the file and shows a success message
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User does not select any contact
++
+[none]
+** 1a1. ContactSails shows an error message
++
+Use case ends.
+
+[none]
+* 1b. User makes an invalid selection
++
+[none]
+** 1b1. ContactSails shows an error message
++
+Use case ends.
+
+[none]
+* 4a. ContactSails is unable to create the file with selected contacts
++
+[none]
+** 4a1. ContactSails shows an error message
++
+Use case resumes at step 1.
+
+[none]
+* 6a. ContactSails is unable to save the file in the specified location
++
+[none]
+** 6a1. ContactSails shows an error message
++
+Use case resumes at step 5.
+
+[none]
+* *a. User cancels the operation at any time
++
+[none]
+** *a1. ContactSails shows an error message.
++
+Use case ends.
+
+[discrete]
+=== Use case: Store a local backup of contacts
+
+*MSS*
+
+1.  User requests to create a backup file containing all the data stored in the application
+2.  ContactSails creates the file and adds all the application data
+3.  User specifies location of file to be saved
+4.  ContactSails saves the file in the location specified and shows a success message
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. ContactSails is unable to create the file
++
+[none]
+** 2a1. ContactSails shows an error message
++
+Use case ends.
+
+[none]
+* 4a. ContactSails is unable to save the file in the specified location
++
+[none]
+** 4a1. ContactSails shows an error message
++
+Use case resumes at step 3.
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-_{More to be added}_
+.  The ContactSails should be usable even for users without experience in social media marketing.
```

Comment:
errr... don't you think this is unrealistic? 
you are targetting for SMM, but you want your product to be used by people who have no experience in SMM?